src/hotspot/cpu/s390/templateTable_s390.cpp

```diff
@@ -251,34 +251,43 @@ void TemplateTable::patch_bytecode(Bytecodes::Code bc,
         // additional, required work.
         assert(byte_no == f1_byte || byte_no == f2_byte, "byte_no out of range");
         assert(load_bc_into_bc_reg, "we use bc_reg as temp");
-        __ get_cache_and_index_and_bytecode_at_bcp(Z_R1_scratch, bc_reg,
-                                                   temp_reg, byte_no, 1);
+
+        // Both registers are block-local temp regs. Their contents before and after is not used.
+        Register index = bc_reg;
+        Register cache = temp_reg;
+
+        __ load_field_entry(cache, index);
         __ load_const_optimized(bc_reg, bc);
-        __ compareU32_and_branch(temp_reg, (intptr_t)0,
-                                 Assembler::bcondZero, L_patch_done);
+
+        if (byte_no == f1_byte) {
+          __ z_cli(Address(cache, in_bytes(ResolvedFieldEntry::get_code_offset())), 0);
+        } else {
+          __ z_cli(Address(cache, in_bytes(ResolvedFieldEntry::put_code_offset())), 0);
+        }
+        __ z_bre(L_patch_done);
       }
       break;
     default:
       assert(byte_no == -1, "sanity");
-      // The pair bytecodes have already done the load.
+      // The bytecode pair may have already performed the load.
       if (load_bc_into_bc_reg) {
         __ load_const_optimized(bc_reg, bc);
       }
       break;
   }
 
   if (JvmtiExport::can_post_breakpoint()) {
-
-    Label   L_fast_patch;
+    NearLabel L_fast_patch;
```

Comment:
Hi @RealLucy , @TheRealMDoerr 

Do you suggest to revert it back, so that situation like [JDK-8317851](https://bugs.openjdk.org/browse/JDK-8317851), could be avoided ? 

Thanks,

Reply:
No.